qdrant_client/qdrant_client.py

```diff
@@ -1,73 +1,35 @@
-import math
-from itertools import count, islice
-from typing import Optional, Iterable, Dict, List, Union, Any
+import collections
+from typing import Optional, Iterable, Dict, List, Union
 
 import numpy as np
 from tqdm import tqdm
 
 from qdrant_client.http import SyncApis
-from qdrant_client.parallel_processor import Worker, ParallelWorkerPool
-from qdrant_client.http.models import PointsBatch, Batch, Filter, SearchParams, SearchRequest, Distance, \
+from qdrant_client.http.models import Filter, SearchParams, SearchRequest, Distance, \
     HnswConfigDiff, OptimizersConfigDiff, WalConfigDiff, CreateCollection, CreateFieldIndex, PointRequest, \
-    PayloadInterface, GeoPoint, PayloadInterfaceStrictOneOf, PayloadInterfaceStrictOneOf1, \
-    PayloadInterfaceStrictOneOf2, PayloadInterfaceStrictOneOf3, ExtendedPointId
-
-
-def iter_batch(iterable, size) -> Iterable:
-    """
-    >>> list(iter_batch([1,2,3,4,5], 3))
-    [[1, 2, 3], [4, 5]]
-    """
-    source_iter = iter(iterable)
-    while source_iter:
-        b = list(islice(source_iter, size))
-        if len(b) == 0:
-            break
-        yield b
-
-
-def _upload_batch(openapi_client: SyncApis, collection_name: str, batch) -> bool:
-    ids_batch, vectors_batch, payload_batch = batch
-
-    openapi_client.points_api.upsert_points(
-        collection_name=collection_name,
-        point_insert_operations=PointsBatch(
-            batch=Batch(
-                ids=ids_batch,
-                payloads=payload_batch,
-                vectors=vectors_batch
-            )
-        )
-    )
-    return True
+    PayloadInterface, ExtendedPointId
+from qdrant_client.parallel_processor import ParallelWorkerPool
+from qdrant_client.uploader.grpc_uploader import GrpcBatchUploader
+from qdrant_client.uploader.rest_uploader import json_to_payload, RestBatchUploader
 
 
 class QdrantClient:
     # If True - single-value payload arrays will be replaced with just values.
     # Warn: Deprecated
     unwrap_payload = False
 
-    def __init__(self, host="localhost", port=6333, **kwargs):
+    def __init__(self,
+                 host="localhost",
+                 port=6333,
+                 grpc_port=6334,
+                 prefer_grpc=False,
```

Comment:
Currently, it seems that GRPC is only used in `upload` method. Since we confirmed that GRPC provides a huge speedup in qdrant/qdrant#351, I believe we can invest more to support GRPC as a first-class citizen in the library. 

Things we can improve:
- Create GRPC client and initialize clients with it as class properties: It may be something like `QdrantClient.grpc.collections`, `QdrantClient.grpc.points` as opposed to `QdrantClient.http`.
- Or, we can separate client classes, but I guess the first one is more user-friendly.

Reply:
will do in later PRs